eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -399,22 +399,15 @@ object OriginSerializer extends MinimalSerializer({
   })
 })
 
-object GlobalBalanceSerializer extends MinimalSerializer({
-  case o: GlobalBalance =>
-    val formats = DefaultFormats + ByteVector32KeySerializer + BtcSerializer + SatoshiSerializer
-    JObject(JField("total", JDecimal(o.total.toDouble))) merge Extraction.decompose(o)(formats)
-})
-
 // @formatter:off
+private case class GlobalBalanceJson(total: Btc, onChain: CorrectedOnChainBalance, offChain: OffChainBalance)
+object GlobalBalanceSerializer extends ConvertClassSerializer[GlobalBalance](b => GlobalBalanceJson(b.total, b.onChain, b.offChain))
+
 private case class PeerInfoJson(nodeId: PublicKey, state: String, address: Option[InetSocketAddress], channels: Int)
 object PeerInfoSerializer extends ConvertClassSerializer[Peer.PeerInfo](peerInfo => PeerInfoJson(peerInfo.nodeId, peerInfo.state.toString, peerInfo.address, peerInfo.channels))
-// @formatter:on
 
-// @formatter:off
 private[json] case class MessageReceivedJson(pathId: Option[ByteVector], replyPath: Option[BlindedRoute], unknownTlvs: Map[String, ByteVector])
-object OnionMessageReceivedSerializer extends ConvertClassSerializer[OnionMessages.ReceiveMessage]({ m: OnionMessages.ReceiveMessage =>
-  MessageReceivedJson(m.pathId, m.finalPayload.replyPath.map(_.blindedRoute), m.finalPayload.records.unknown.map(tlv => (tlv.tag.toString -> tlv.value)).toMap)
-})
+object OnionMessageReceivedSerializer extends ConvertClassSerializer[OnionMessages.ReceiveMessage](m => MessageReceivedJson(m.pathId, m.finalPayload.replyPath.map(_.blindedRoute), m.finalPayload.records.unknown.map(tlv => tlv.tag.toString -> tlv.value).toMap))
```

Comment:
Side note : not sure why the internal case class one line above was defined with `private[json]`.

Reply:
It's because it's needed for the websocket type hints: https://github.com/ACINQ/eclair/blob/535daec0657acf2e05f5c8fca046ae9ab3f70434/eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala#L455